src/common/prometheus_client.cc

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "knowhere/comp/prometheus_client.h"
+
+namespace knowhere {
+
+/*****************************************************************************/
+// prometheus metrics
+PrometheusClient* prometheusClient = new PrometheusClient();
+
+/*******************************************************************************
+ * Examples:
+ *
+ * PROMETHEUS_GAUGE(knowhere_search_gauge,
+ *                  knowhere_search_gauge_family,
+ *                  "knowhere_search_gauge",
+ *                  "total search gauge in knowhere",
+ *                  *prometheusClient->GetRegistry())
+ *
+ * PROMETHEUS_COUNTER(knowhere_search_counter,
+ *                    knowhere_search_counter_family,
+ *                    "knowhere_search_counter",
+ *                    "total search counter in knowhere",
+ *                    *prometheusClient->GetRegistry())
+ *
+ * Quantiles quantiles = {{0.95, 0.01}, {0.9, 0.02}, {0.8, 0.1}};
```

Comment:
Is it possible use raw value directly in the declaration?

Reply:
updated